awx/ui_next/src/screens/Inventory/InventoryList/useWsInventories.js

```diff
@@ -64,16 +75,41 @@ export default function useWsProjects(
         return;
       }
 
-      if (!['pending', 'waiting', 'running'].includes(lastMessage.status)) {
-        enqueueId(lastMessage.inventory_id);
-        return;
-      }
-
       const inventory = inventories[index];
       const updatedInventory = {
         ...inventory,
-        isSourceSyncRunning: true,
       };
+
+      if (lastMessage.group_name === 'inventories') {
+        if (lastMessage.status === 'pending_deletion') {
+          updatedInventory.pending_deletion = true;
+        } else if (lastMessage.status === 'deleted') {
+          if (inventories.length === 1) {
+            const params = parseQueryString(qsConfig, location.search);
+            if (params.page > 1) {
+              const newParams = encodeNonDefaultQueryString(
+                qsConfig,
+                replaceParams(params, {
+                  page: params.page - 1,
+                })
+              );
+              history.push(`${location.pathname}?${newParams}`);
+            }
+          } else {
+            fetchInventories();
+          }
+        } else {
+          return;
+        }
+      } else {
+        if (!['pending', 'waiting', 'running'].includes(lastMessage.status)) {
+          enqueueId(lastMessage.inventory_id);
+          return;
+        }
+
+        updatedInventory.isSourceSyncRunning = true;
+      }
```

Comment:
A lot of nested if/else can be tricky to to follow. Maybe move part of this into a helper function(s), or maybe handle cases one at a time, returning out of this function when possible (so you don't need an `else`)?